File: core/src/main/java/org/stepik/plugin/projectWizard/ui/CourseDescriptionPane.java
package org.stepik.plugin.projectWizard.ui;

import com.intellij.ide.BrowserUtil;
import com.intellij.ui.HyperlinkAdapter;

import javax.swing.*;
import javax.swing.event.HyperlinkEvent;

/**
 * @author meanmail
 */
public class CourseDescriptionPane extends JTextPane {
    public CourseDescriptionPane() {
        setEditable(false);
        setContentType("text/html");
        addHyperlinkListener(new HyperlinkAdapter() {
            @Override
            protected void hyperlinkActivated(final HyperlinkEvent e) {
                BrowserUtil.browse(e.getURL());
            }
        });
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/projectWizard/ui/CourseDescriptionPane.kt
package org.stepik.core.projectWizard.ui

import com.intellij.ide.BrowserUtil
import com.intellij.ui.HyperlinkAdapter
import javax.swing.JTextPane
import javax.swing.event.HyperlinkEvent


class CourseDescriptionPane : JTextPane() {
    init {
        isEditable = false
        contentType = "text/html"
        addHyperlinkListener(object : HyperlinkAdapter() {
            override fun hyperlinkActivated(e: HyperlinkEvent) {
                BrowserUtil.browse(e.url)
            }
        })
    }
}
